### PR TITLE
VRP - fix global VRF being NULL and not '' for cbgp, support for NetEngine devices

### PR DESCRIPTION
**Pull request by PipoCanaja**
Fix for:
https://community.librenms.org/t/bgp-peers-routes-adviertised-received-info-missing

cbgp DB entry for GLOBAL peer (not in VRF) was set with context = '' instead of context=NULL

Also fixes regex to match "hardware" detection when model contains a space. 

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/definitions/discovery/vrp.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 mib: HUAWEI-WLAN-CONFIGURATION-MIB:HUAWEI-WAN-MIB:HUAWEI-ENTITY-EXTENT-MIB:HUAWEI-ENERGYMNGT-MIB:HUAWEI-STACK-MIB
 modules:
     os:
-        sysDescr_regex: '/\((?<hardware>\S+) (?<version>V[0-9]{3}R[0-9]{3}[0-9A-Z]+)/'
+        sysDescr_regex: '/\((?<hardware>[^ V]+( (?!V)[^ V]*)*) (?<version>V[0-9]{3}R[0-9]{3}[0-9A-Z]+)/'
```
Comment (Jellyfrog) [inline includes/definitions/discovery/vrp.yaml:4]: Maybe a simpler fix is something like `\((?<hardware>[^V\)]+)(?<version>V[^)]+)\)`
Havent checked if all devices follow the pattern of `(*hardware* V*version*)` tho

**Comment by PipoCanaja** [comment]
Patch was tested by requester in his LibreNMS deployment. Works as expected. Will do some more tests on my deployments with other devices to check for regressions if any.

File: includes/definitions/discovery/vrp.yaml (hunk 1 above)
Comment (murrant) [reply includes/definitions/discovery/vrp.yaml:4]: how about .*

Comment (murrant) [reply includes/definitions/discovery/vrp.yaml:4]: `\((?<hardware>[^)]+) (?<version>V[0-9]{3}R[0-9]{3}[0-9A-Z]+)`

Comment (murrant) [reply includes/definitions/discovery/vrp.yaml:4]: A little more generic version of Jellyfrog's, `\((?<hardware>[^\)]+)(?<version>V[^)]+)\)`  146 steps

Comment (PipoCanaja) [reply includes/definitions/discovery/vrp.yaml]: Right, both of you ! Never, ever, play with regexp at "farTooLate oclock"... running the checks locally succeed with test data. Should be all right with the PR.

**Review by murrant — APPROVED** [review]

**Pull request merged**